src/Service/SymfonyEventDispatcherEmitter.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace League\Bundle\OAuth2ServerBundle\Service;
+
+use League\Event\EmitterInterface;
+use League\Event\Event;
+use League\Event\EventInterface;
+use League\Event\GeneratorInterface;
+use League\Event\ListenerProviderInterface;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+final class SymfonyEventDispatcherEmitter implements EmitterInterface
+{
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    public function __construct(EventDispatcherInterface $eventDispatcher)
+    {
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function removeListener($event, $listener)
+    {
+        // nothing to do here
+    }
+
+    public function useListenerProvider(ListenerProviderInterface $provider)
+    {
+        // nothing to do here
+    }
+
+    public function removeAllListeners($event)
+    {
+        // nothing to do here
+    }
+
+    public function hasListeners($event)
+    {
+        // nothing to do here
+    }
+
+    public function getListeners($event)
+    {
+        // nothing to do here
+    }
+
+    public function emit($event): EventInterface
+    {
+        $preparedEvent = $this->prepareEvent($event);
+
+        $this->eventDispatcher->dispatch($preparedEvent, $preparedEvent->getName());
+
+        return $preparedEvent;
+    }
+
+    public function emitBatch(array $events): array
+    {
+        $results = [];
+
+        foreach ($events as $event) {
+            $results[] = $this->emit($event);
+        }
+
+        return $results;
```

Comment:
Maybe use array_map ?
```suggestion
         return array_map(function ($event) {
            return $this->emit($event);
        }, $events);
```